zcncore/wallet.go

```diff
@@ -289,15 +290,35 @@ func CloseLog() {
 
 // Init inializes the SDK with miner, sharder and signature scheme provided in
 // configuration provided in JSON format
-func Init(c string) error {
-	err := json.Unmarshal([]byte(c), &_config.chain)
+// It is used for 0proxy, 0box, 0explorer, andorid, ios : walletJSON is ChainConfig
+//	 {
+//      "chain_id":"0afc093ffb509f059c55478bc1a60351cef7b4e9c008a53a6cc8241ca8617dfe",
+//		"signature_scheme" : "bls0chain",
+//		"block_worker" : "https://dev.0chain.net/dns",
+// 		"min_submit" : 50,
+//		"min_confirmation" : 50,
+//		"confirmation_chain_length" : 3,
+//		"num_keys" : 1,
+//		"eth_node" : "https://ropsten.infura.io/v3/f0a254d8d18b4749bd8540da63b3292b"
+//	 }
+func Init(chainConfigJSON string) error {
+	err := json.Unmarshal([]byte(chainConfigJSON), &_config.chain)
 	if err == nil {
 		// Check signature scheme is supported
 		if _config.chain.SignatureScheme != "ed25519" && _config.chain.SignatureScheme != "bls0chain" {
 			return errors.New("", "invalid/unsupported signature scheme")
 		}
 
-		err := UpdateNetworkDetails()
+		// try to initiazlie conf/Config
+		reader, err := conf.NewReaderFromJSON(chainConfigJSON)
```

Comment:
This code logic looks duplicated with code logic above to unmarshal config
`err := json.Unmarshal([]byte(chainConfigJSON), &_config.chain)`

Reply:
can we just move the checking on `err := json.Unmarshal([]byte(chainConfigJSON), &_config.chain)` along with signature checking into `conf.NewReaderFromJSON(chainConfigJSON)` instead?